File: iosApp/iosApp/font/model/PlatformFontPathProviderImpl.swift
//
//  PlatformFontPathProviderImpl.swift
//  iosApp
//
//  Created by vlad on 14/7/21.
//

import Foundation
import shared

class PlatformFontPathProviderImpl : PlatformFontPathProvider {
    private let robotoFont = PredefinedFontPath(path: FontsManager.Companion().ROBOTO_FONT_PATH, displayName: FontsManager.Companion().ROBOTO_FONT_PATH.capitalized,
                                                regularId: FontResource(fontName: "roboto-regular.ttf", bundle: Bundle.main), italicId: FontResource(fontName: "roboto-italic.ttf", bundle: Bundle.main),
                                                lightId: FontResource(fontName: "roboto-light.ttf", bundle: Bundle.main), boldId: FontResource(fontName: "roboto-bold.ttf", bundle: Bundle.main), forPremium: false
    )
    private let sfProFont = PredefinedFontPath(path: FontsManager.Companion().SF_FONT_PATH,
                                               displayName: FontsManager.Companion().SF_FONT_NAME, regularId: nil, italicId: nil, lightId: nil, boldId: nil, forPremium: false)
    
    
    func getRobotoFont() -> PredefinedFontPath {
        return robotoFont
    }
    
    func getSfProFont() -> PredefinedFontPath {
        return sfProFont
    }
    
    func defaultFont() -> PredefinedFontPath {
        return sfProFont
    }
}

File: shared/src/androidMain/kotlin/app/inspiry/font/provider/PlatformFontPathProviderImpl.kt
package app.inspiry.font.provider

import app.inspiry.font.model.PredefinedFontPath
import app.inspiry.projectutils.R
import dev.icerock.moko.resources.FontResource

class PlatformFontPathProviderImpl : PlatformFontPathProvider {

    private val _robotoFont = PredefinedFontPath(FontsManager.ROBOTO_FONT_PATH)
    private val sfFont = PredefinedFontPath(
        FontsManager.SF_FONT_PATH,
        FontsManager.SF_FONT_NAME,
        regularId = FontResource(R.font.sf_pro_display_regular),
        italicId = FontResource(R.font.sf_pro_display_italic),
        boldId = FontResource(R.font.sf_pro_display_bold),
        lightId = FontResource(R.font.sf_pro_display_light)
    )

    override fun getRobotoFont(): PredefinedFontPath {
        return _robotoFont
    }

    override fun getSfProFont(): PredefinedFontPath {
        return sfFont
    }

    override fun defaultFont(): PredefinedFontPath {
        return getRobotoFont()
    }
}